File: changelog_builder/commit_processor/commit_ignore.py
import re


class CommitIgnore:
    def __init__(self, config):
        self.config = config

    def process(self, commits):
        if len(self.config['ignore_comments']) == 0:
            return commits

        valid_commits = {}
        for ignore_comment in self.config['ignore_comments']:
            for key in set(commits):
                match = re.search(ignore_comment, commits[key])
                if None == match:
                    valid_commits[key] = commits[key]

        return valid_commits

File: lib/processor/commitignore.js
var config = require('../../config');

function CommitIgnore() {

}

CommitIgnore.prototype.process = function(commits) {
	if (config.ignore_comments.length === 0) return commits;

	for (var i in config.ignore_comments) {
		var ignoreComment = config.ignore_comments[i];
		for (var j in commits) {
			if (commits.hasOwnProperty(j)) {
				var commit = commits[j];
				if (commit.search(ignoreComment) != -1) {
					delete commits[j];
				}
			}
		}
	}

	return commits;
}

module.exports = CommitIgnore;
